### ISSUE TITLE
scatter softmax output 'nan'

### ISSUE THREAD

**Issue by dytan** [rusty1s/pytorch_geometric#53]
https://github.com/rusty1s/pytorch_geometric/blob/83b451f9ecf4d1fc7c674c6461937ffaae4135ac/torch_geometric/utils/softmax.py#L9

Thanks for the fancy library.
On the use of GatConv, it produce 'nan' frequently, after checkout the code, found this line doing exp straightly cause overflow, should it be better to sub the max beforehand.

    row_max, _ = scatter_max(src, index, dim=0, dim_size=num_nodes)
    src = src - row_max[index]
    out = src.exp()

**Comment by rusty1s** [comment]
Thanks a lot! Fixed in master!
